### PR TITLE
[DEMO] upload files from FileStorage to ToolStorage

### LINKED ISSUE #1771: [DEMO] upload files from FileStorage to ToolStorage

### PR DESCRIPTION
**Pull request by sanyavertolet**
This PR closes #1771

### What's done:
 * Improved `simpleFileUploader's` usage to support file uploading from `FileStorage` to `ToolStorage`
 * Renamed `GithubDownloadToolService` to `DownloadToolService`
 * Implemented file downloading from `save-backend` to `save-demo` (done as a part of `DownloadToolService`)
 * Added `ERROR` status to `DemoStatus`
 * Added dependency table to `save-demo` in order to keep records about tool's dependencies

### CODE REVIEW

**Review by 0x6675636b796f75676974687562 — APPROVED** [review]

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/projects/ProjectDemoMenu.kt [hunk 1]
```diff
@@ -95,12 +94,12 @@ val projectDemoMenu: FC<ProjectDemoMenuProps> = FC { props ->
...
         } else if (infoResponse.status != 404.toShort()) {
             props.updateErrorMessage(infoResponse.statusText, infoResponse.unpackMessage())
-            setDemoStatus(DemoStatus.NOT_CREATED)
+            setDemoStatus(DemoStatus.ERROR)
         }
     }
 
     useOnce {
-        window.alert("This is just a preview, nothing on this view works for now.")
+        // window.alert("This is just a preview, nothing on this view works for now.")
```
Comment (nulls) [inline save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/projects/ProjectDemoMenu.kt:102]: a comment code

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/projects/ProjectDemoMenu.kt [hunk 2]
```diff
@@ -222,7 +229,7 @@ private fun ChildrenBuilder.renderFileUploading(
                 autoComplete = AutoComplete.off
                 placeholder = "Release tag"
                 value = demoDto.vcsTagName
-                disabled = demoStatus != DemoStatus.NOT_CREATED && demoStatus != DemoStatus.STOPPED
+                disabled = true
```
Comment (nulls) [inline save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/projects/ProjectDemoMenu.kt:232]: is it a temporary disabled?

File: save-cloud-common/src/jvmMain/kotlin/com/saveourtool/save/utils/KotlinUtilsJvm.kt [hunk 3]
```diff
@@ -7,3 +7,4 @@ package com.saveourtool.save.utils
 import org.springframework.http.ResponseEntity
 
 typealias EmptyResponse = ResponseEntity<Void>
+typealias StringResponse = ResponseEntity<String>
```
Comment (nulls) [inline save-cloud-common/src/jvmMain/kotlin/com/saveourtool/save/utils/KotlinUtilsJvm.kt:10]: we have `StrungResponse` in `com/saveourtool/save/backend/SaveApplication.kt:11`, can you please replace old one by a new one introduced by you?

**Review by nulls — APPROVED** [review]

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/projects/ProjectDemoMenu.kt (hunk 2 above)
Comment (sanyavertolet) [reply save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/projects/ProjectDemoMenu.kt:232]: Yes, it is disabled until we allow to choose between GitHub download and manual download

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/projects/ProjectDemoMenu.kt (hunk 1 above)
Comment (sanyavertolet) [reply save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/projects/ProjectDemoMenu.kt:102]: Oops, shouldn't be commented, will fix it

File: save-cloud-common/src/jvmMain/kotlin/com/saveourtool/save/utils/KotlinUtilsJvm.kt (hunk 3 above)
Comment (sanyavertolet) [reply save-cloud-common/src/jvmMain/kotlin/com/saveourtool/save/utils/KotlinUtilsJvm.kt:10]: Sure

Comment (sanyavertolet) [reply save-cloud-common/src/jvmMain/kotlin/com/saveourtool/save/utils/KotlinUtilsJvm.kt:10]: Done!

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/projects/ProjectDemoMenu.kt (hunk 1 above)
Comment (sanyavertolet) [reply save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/projects/ProjectDemoMenu.kt]: Uncommented the alert

**Pull request merged**